tests/rust/wasm32-wasip3/src/bin/filesystem-unlink-errors.rs

```diff
@@ -0,0 +1,54 @@
+use std::process;
+extern crate wit_bindgen;
+
+wit_bindgen::generate!({
+    inline: r"
+  package test:test;
+
+  world test {
+      include wasi:filesystem/imports@0.3.0-rc-2025-09-16;
+      include wasi:cli/command@0.3.0-rc-2025-09-16;
+  }
+",
+    additional_derives: [PartialEq, Eq, Hash, Clone],
+    // Work around https://github.com/bytecodealliance/wasm-tools/issues/2285.
+    features:["clocks-timezone"],
+    generate_all
+});
+
+use wasi::filesystem::types::Descriptor;
+use wasi::filesystem::types::ErrorCode;
+
+async fn test_unlink_errors(dir: &Descriptor) {
+    let rm = |path: &str| dir.unlink_file_at(path.to_string());
+    assert_eq!(rm("").await, Err(ErrorCode::NoEntry));
+    assert_eq!(rm(".").await, Err(ErrorCode::IsDirectory));
```

Comment:
MacOS error:
Test filesystem-unlink-errors failed
  [exit_code] 0 == 134
STDOUT:

STDERR:

thread '<unnamed>' (1) panicked at src/bin/filesystem-unlink-errors.rs:25:5:
assertion `left == right` failed
  left: Err(ErrorCode { code: 30, name: "not-permitted", message: "Operation not permitted, similar to `EPERM` in POSIX." })
 right: Err(ErrorCode { code: 13, name: "is-directory", message: "Is a directory, similar to `EISDIR` in POSIX." })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: failed to run main module `tests/rust/testsuite/wasm32-wasip3/filesystem-unlink-errors.wasm`